src/service/resources-impl.js

```diff
@@ -742,7 +742,7 @@ export class Resources {
         if (diff == 0) {
```

Comment:
This conditional can be joined with the following (you'll have to move `resize` declaration up). With that, we can remove the `request.callback` stuff here, too.